src/app/components/applications/Faucet/RequestForSsv/RequestForSsv.tsx

```diff
@@ -15,83 +15,88 @@ import WalletStore from '~app/common/stores/applications/Faucet/Wallet.store';
 import ApplicationStore from '~app/common/stores/applications/SsvWeb/Application.store';
 import { useStyles } from '~app/components/applications/Faucet/RequestForSsv/RequestForSsv.styles';
 
+type CaptchaThemeType = 'dark' | 'light';
+
 const RequestForSsv = () => {
-    const stores = useStores();
-    const classes = useStyles();
-    const navigate = useNavigate();
-    const captchaRef = useRef(null);
-    const faucetStore: FaucetStore = stores.Faucet;
-    const walletStore: WalletStore = stores.Wallet;
-    const [error, setError] = useState('');
-    const [disabled, setDisabled] = useState(true);
-    const applicationStore: ApplicationStore = stores.Application;
-    const [buttonText, setButtonText] = useState('Request');
-    const [reachedMaxTransactionPerDay, setReachedMaxTransactionPerDay] = useState(false);
+  const stores = useStores();
+  const classes = useStyles();
+  const navigate = useNavigate();
+  const captchaRef = useRef(null);
+  const faucetStore: FaucetStore = stores.Faucet;
+  const walletStore: WalletStore = stores.Wallet;
+  const [error, setError] = useState('');
+  const [disabled, setDisabled] = useState(true);
+  const applicationStore: ApplicationStore = stores.Application;
+  const [buttonText, setButtonText] = useState('Request');
+  const [reachedMaxTransactionPerDay, setReachedMaxTransactionPerDay] = useState(false);
+  const captchaTheme = applicationStore.theme.darkMode ? config.THEMES.DARK : config.THEMES.LIGHT;
 
-    useEffect(() => {
-        setError('');
-        setDisabled(true);
-        setButtonText('Request');
-    }, [walletStore.accountAddress]);
+  useEffect(() => {
+    setError('');
+    setDisabled(true);
+    setButtonText('Request');
+  }, [walletStore.accountAddress]);
 
-    const requestForSSV = async () => {
-        setError('');
-        setButtonText('Requesting...');
-        applicationStore.setIsLoading(true);
-        const response = await faucetStore.registerNewTransaction();
-        if (!response.status) {
-            if (response.type === 2) {
-                applicationStore.setIsLoading(false);
-                navigate(config.routes.FAUCET.DEPLETED);
-            } else {
-                setError('Reached Max Transactions Per Day');
-                setReachedMaxTransactionPerDay(true);
-                applicationStore.setIsLoading(false);
-                setButtonText('Request');
-            }
-            return;
-        }
+  const requestForSSV = async () => {
+    setError('');
+    setButtonText('Requesting...');
+    applicationStore.setIsLoading(true);
+    const response = await faucetStore.registerNewTransaction();
+    if (!response.status) {
+      if (response.type === 2) {
```

Comment:
I don't think that using number for api responses is clear. I understand that it is due to the implementation registerNewTransaction():

} catch (e: any) {
            return { status: false, type: e.response.data.error.message === 'Reached max transactions per day' ? 1 : 2 };
        }

I think we should have a facuetService which calls faucet urls parses its response and errors. We should have faucet errors as constants FaucetError.MAX_DAILY_TX_REACHED . In registerNewTransaction() we check:
type: e.response.data.error.message === 'Reached max transactions per day'. 

then we setError here on line 50 with same text.

Faucet service isn't critical but please add the error as constant.

Also probably it should be "response.type === 1"?

Reply:
Type 1 means that we have reached the maximum number of transactions per day. Type 2 means the faucet is depleted.